src/oauth.tsx

```diff
@@ -322,7 +322,7 @@ const tokenRequestSchema = z.object({
   code: z.string().optional(),
   client_id: z.string(),
   client_secret: z.string(),
-  redirect_uri: z.string().url(),
+  redirect_uri: z.string().url().optional(),
```

Comment:
Another option might be a discriminated union, so "these are the parameters required for authorization_code and these are required for client_credentials"

Technically `client_secret` is also optional once you support public clients, not just confidential clients.